picard/file.py

```diff
@@ -285,6 +289,7 @@ def _save_and_rename(self, old_filename, metadata):
         # Save cover art images
         if config.setting["save_images_to_files"]:
             self._save_images(os.path.dirname(new_filename), metadata)
+        run_file_post_save_processors(self, new_filename)
```

Comment:
I'm not sure if this is the correct place for this - https://github.com/metabrainz/picard/blob/a1a79658bf58d150c927e7ff03b96c8e938d0ae6/picard/file.py#L213 calls `_saving_finished` *after* `_save_and_rename` and `_saving_finished` still does a lot of work in the non-error case, including updating `self.filename` and some metadata. If we want to merge this in its current form, the documentation needs to clearly list which updates are guaranteed to have happened at the point the post-save-processor runs and which are still not done. In particular the not-yet-updated filename of the `File` object in both the `filename` attribute and the metadata is super problematic for the automatic-submission-to-acousticbrainz usecase.

Reply:
The file is actually saved in `_save_and_rename`, but i agree it makes more sense to call the hook in `_saving_finished`, and just pass file object.

Reply:
I did that in 5e93e1529fb09e, does it look better ?